packages/synapse-react-client/src/components/ChangePassword/ChangePassword.integration.test.tsx

```diff
@@ -0,0 +1,440 @@
+
+    await userEvent.click(submitButton)
+
+    await waitFor(() => {
+      const alert = screen.getByRole('alert')
+      within(alert).getByText(errorMessage)
+
+      expect(mockDisplayToast).not.toHaveBeenCalled()
+      expect(currentPasswordField).toHaveValue(currentPassword)
+      expect(newPasswordField).toHaveValue(newPassword)
+      expect(confirmPasswordField).toHaveValue(newPassword)
+
+      expect(changePasswordSpy).toHaveBeenCalledTimes(1)
+      expect(changePasswordSpy).toHaveBeenCalledWith({
+        concreteType:
+          'org.sagebionetworks.repo.model.auth.ChangePasswordWithCurrentPassword',
+        username: mockUserProfileData.userName,
+        newPassword: newPassword,
+        currentPassword: currentPassword,
+      })
+    })
+  })
+
+  it('supports entering a TOTP if the server requires 2FA', async () => {
+    const userId = MOCK_USER_ID
+    const twoFaToken = 'mock-2fa-token'
+    server.use(
+      // The first call will indicate that 2FA is required
+      getRequires2FAChangePasswordHandler(
+        getEndpoint(BackendDestinationEnum.REPO_ENDPOINT),
+        userId,
+        twoFaToken,
+      ),
+      // Update the mock server so the second request will succeed
+      getSuccessfulChangePasswordHandler(
+        getEndpoint(BackendDestinationEnum.REPO_ENDPOINT),
+      ),
+    )
+
+    const currentPassword = 'currentPassword'
+    const newPassword = 'newPassword'
+
+    const {
+      user,
+      currentPasswordField,
+      newPasswordField,
+      confirmPasswordField,
+      submitButton,
+    } = setUp()
+
+    await user.type(currentPasswordField, currentPassword)
+    await user.type(newPasswordField, newPassword)
+    await user.type(confirmPasswordField, newPassword)
+
+    await userEvent.click(submitButton)
+
+    // TOTP form should pop up
+    let otpInputs: HTMLElement[] = []
+    await waitFor(() => {
+      otpInputs = screen.getAllByRole('textbox')
+      expect(otpInputs).toHaveLength(6)
+
+      const alert = screen.getByRole('alert')
+      within(alert).getByText(
+        'Two-factor authentication is required to change your password.',
+      )
+
+      expect(mockDisplayToast).not.toHaveBeenCalled()
+      expect(currentPasswordField).not.toBeInTheDocument()
+      expect(newPasswordField).not.toBeInTheDocument()
+      expect(confirmPasswordField).not.toBeInTheDocument()
+
+      expect(changePasswordSpy).toHaveBeenCalledTimes(1)
+      expect(changePasswordSpy).toHaveBeenCalledWith({
+        concreteType:
+          'org.sagebionetworks.repo.model.auth.ChangePasswordWithCurrentPassword',
+        username: mockUserProfileData.userName,
+        newPassword: newPassword,
+        currentPassword: currentPassword,
+      })
+    })
+
+    // Type the code. Once the code is entered, the form should submit automatically
+    for (let i = 0; i < otpInputs.length; i++) {
+      await userEvent.type(otpInputs[i], String(i + 1))
```

Comment:
Is it possible to use `user` here as well?

```suggestion
      await user.type(otpInputs[i], String(i + 1))
```